src/celldega/pre/merge_segmentations.py

```diff
@@ -0,0 +1,823 @@
+    ]
+
+    indices_intersecting_cells_before_harmonization = (
+        intersecting_cells_before_harmonization.index.tolist()
+    )
+    intersecting_cell_pairs_before_harmonization = (
+        intersecting_cells_before_harmonization.sindex.query(
+            intersecting_cells_before_harmonization.geometry, predicate="intersects"
+        )
+    )
+
+    intersecting_cell_pairs_before_harmonization_DF = pd.DataFrame(
+        intersecting_cell_pairs_before_harmonization
+    ).T
+    intersecting_cell_pairs_before_harmonization_DF.columns = ["cell_id_1", "cell_id_2"]
+
+    intersecting_cell_pairs_before_harmonization_DF["cell_id_1"] = (
+        intersecting_cells_before_harmonization.index[
+            intersecting_cell_pairs_before_harmonization_DF["cell_id_1"]
+        ].values
+    )
+    intersecting_cell_pairs_before_harmonization_DF["cell_id_2"] = (
+        intersecting_cells_before_harmonization.index[
+            intersecting_cell_pairs_before_harmonization_DF["cell_id_2"]
+        ].values
+    )
+    intersecting_cell_pairs_before_harmonization_DF["cell_id_1_technology"] = (
+        intersecting_cells_before_harmonization.technology[
+            intersecting_cell_pairs_before_harmonization_DF["cell_id_1"]
+        ].values
+    )
+    intersecting_cell_pairs_before_harmonization_DF["cell_id_2_technology"] = (
+        intersecting_cells_before_harmonization.technology[
+            intersecting_cell_pairs_before_harmonization_DF["cell_id_2"]
+        ].values
+    )
+
+    intersecting_cell_pairs_before_harmonization_DF = (
+        intersecting_cell_pairs_before_harmonization_DF[
+            intersecting_cell_pairs_before_harmonization_DF["cell_id_1"]
+            != intersecting_cell_pairs_before_harmonization_DF["cell_id_2"]
+        ]
+    )
+
+    intersecting_cell_pairs_before_harmonization_DF["sorted_cell_id_pair"] = (
+        intersecting_cell_pairs_before_harmonization_DF.apply(
+            lambda row: tuple(sorted([row["cell_id_1"], row["cell_id_2"]])), axis=1
+        )
+    )
+
+    intersecting_cell_pairs_before_harmonization_DF = (
+        intersecting_cell_pairs_before_harmonization_DF.drop_duplicates(
+            subset="sorted_cell_id_pair"
+        )
+    )
+
+    intersecting_cell_pairs_before_harmonization_DF = (
+        intersecting_cell_pairs_before_harmonization_DF.drop(
+            columns=["sorted_cell_id_pair"]
+        )
+    )
+
+    intersecting_cell_pairs_before_harmonization_DF = (
+        intersecting_cell_pairs_before_harmonization_DF[
+            intersecting_cell_pairs_before_harmonization_DF["cell_id_1_technology"]
+            != intersecting_cell_pairs_before_harmonization_DF["cell_id_2_technology"]
+        ]
+    )
+
+    intersecting_cell_pairs_before_harmonization_DF.reset_index(inplace=True)
+    intersecting_cell_pairs_before_harmonization_DF.drop(
+        ["index"], inplace=True, axis=1
+    )
+
```

Comment:
Can we modularize sections of code like this into separate functions so they are easier to work with?